src/app/content/components/BookBanner.tsx

```diff
@@ -192,18 +196,23 @@ export class BookBanner extends Component<PropTypes, {scrollTransition: boolean}
   }
 
   public render() {
-    const {pageNode, book} = this.props;
+    const { pageNode, book } = this.props;
+    const defaultTheme = 'blue' as BookWithOSWebData['theme'];
 
     if (!book || !pageNode) {
       return <BarWrapper colorSchema={undefined} up={false} />;
     }
 
-    const bookUrl = bookDetailsUrl(book);
+    const bookUrl = hasOSWebData(book) ? bookDetailsUrl(book) : '/errors/404';
```

Comment:
`'errors/404'` probably should be some kind of constant, but I'm not familiar with this code base so maybe this is not a case here